cmd/chart-repo/utils.go

```diff
@@ -271,13 +271,23 @@ func parseRepoIndex(body []byte) (*helmrepo.IndexFile, error) {
 	return &index, nil
 }
 
-func chartsFromIndex(index *helmrepo.IndexFile, r repo) []chart {
+func chartsFromIndex(index *helmrepo.IndexFile, r repo, filter *filters) []chart {
 	var charts []chart
 	for _, entry := range index.Entries {
 		if entry[0].GetDeprecated() {
 			log.WithFields(log.Fields{"name": entry[0].GetName()}).Info("skipping deprecated chart")
 			continue
 		}
+
+		if len(filter.Annotations) > 0 ||
+			len(filter.Names) > 0 {
+			if filterEntry(entry[0], filter){
+				charts = append(charts, newChart(entry, r))
```

Comment:
nitpick: it might be simpler to flip this condition and continue if it enters it, and just fallback to the append on line 291 if it matches the filter. i.e. similar to the deprecation condition above

Reply:
Done.  I ended up adding more test data to valid-index.yaml as I removed the ! from  `if !filterEntry(entry[0], filter)` to make the test fail and realized with only two charts of data the test would pass no mater what when filtering.